src/lib/ListMetrics.js

```diff
@@ -3,33 +3,101 @@ import { errors } from 'arsenal';
+     */
+    _getSchemaObject(value) {
+        // key for the schema should not be plural metric type, so remove 's'.
+        let type;
+        if (this.metric === 'buckets') {
+            type = 'bucket';
+        } else if (this.metric === 'accounts') {
+            type = 'accountId';
+        }
+        const obj = {};
+        obj[type] = value;
+        return obj;
+    }
+
+    // Create the metric response object for a given metric.
+    _getMetricRes(value, start, end) {
+        const metricRes = {
+            timeRange: [start, end],
+            storageUtilized: [0, 0],
+            incomingBytes: 0,
+            outgoingBytes: 0,
+            numberOfObjects: [0, 0],
+            operations: {
+                's3:DeleteBucket': 0,
+                's3:DeleteBucketWebsite': 0,
+                's3:ListBucket': 0,
+                's3:GetBucketAcl': 0,
+                's3:GetBucketWebsite': 0,
+                's3:CreateBucket': 0,
+                's3:PutBucketAcl': 0,
+                's3:PutBucketWebsite': 0,
+                's3:PutObject': 0,
+                's3:CopyObject': 0,
+                's3:UploadPart': 0,
+                's3:ListBucketMultipartUploads': 0,
+                's3:ListMultipartUploadParts': 0,
+                's3:InitiateMultipartUpload': 0,
+                's3:CompleteMultipartUpload': 0,
+                's3:AbortMultipartUpload': 0,
+                's3:DeleteObject': 0,
+                's3:MultiObjectDelete': 0,
+                's3:GetObject': 0,
+                's3:GetObjectAcl': 0,
+                's3:PutObjectAcl': 0,
+                's3:HeadBucket': 0,
+                's3:HeadObject': 0,
+            },
+        };
+        const map = {
+            buckets: 'bucketName',
+            accounts: 'accountId',
+        };
+        metricRes[map[this.metric]] = value;
+        return metricRes;
+    }
 
     /**
     * Callback for getting metrics for a list of buckets
-    * @callback Buckets~bucketsMetricsCb
+    * @callback ListMetrics~ListMetricsCb
     * @param {object} err - ArsenalError instance
-    * @param {object[]} buckets - list of objects containing metrics for each
-    * bucket provided in the request
+    * @param {object[]} metric - list of objects containing metrics for each
+    * metric provided in the request
     */
 
     /**
-    * Get metrics for a list of buckets
+    * Get metrics for a list of a metric type
     * @param {utapiRequest} utapiRequest - utapiRequest instance
-    * @param {Buckets~bucketsMetricsCb} cb - callback
+    * @param {ListMetrics~bucketsMetricsCb} cb - callback
     * @return {undefined}
     */
-    static getBucketsMetrics(utapiRequest, cb) {
+    getTypesMetrics(utapiRequest, cb) {
         const log = utapiRequest.getLog();
         const validator = utapiRequest.getValidator();
-        const buckets = validator.get('buckets');
+        const metric = validator.get(this.metric);
```

Comment:
Let's call the list of accounts or buckets as resources and use the language consistently across the file. This will help not to confuse it with metric type.